solr/core/src/java/org/apache/solr/cloud/api/collections/RestoreCmd.java

```diff
@@ -271,7 +278,37 @@ public void call(ClusterState state, ZkNodeProps message, NamedList results) thr
           propMap.put(ASYNC, asyncId);
         }
         ocmh.addPropertyParams(message, propMap);
-        ocmh.addReplica(clusterState, new ZkNodeProps(propMap), new NamedList(), null);
+        final NamedList addResult = new NamedList();
+        ocmh.addReplica(clusterState, new ZkNodeProps(propMap), addResult, () -> {
+          countDownLatch.countDown();
```

Comment:
I think we should move this to the end of the callback, or there can be a race condition